error'ing out when attempting to delete a non-existant file

The tfswitch action exits with an error when the file it tries to delete doesn't exist. This doesn't make any sense.

```
Version: 0.13.1308
Exporting PATH to GITHUB_PATH ..
Unlinking default version of terraform ..
unlink: cannot unlink '/usr/local/bin/terraform': No such file or directory
Error: Process completed with exit code 1.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -80,10 +80,10 @@ jobs:
       run: |
         tfswitch 1.6.6
         terraform --version
-        if [[ -n "$ver" ]]; then
+        echo "Output of terraform_version${{ steps.tfswitch-install.outputs.terraform_version }}"
+        if [[ -n "${{ steps.tfswitch-install.outputs.terraform_version }}" ]]; then
           echo "Output variable is set"
           echo "Terraform ${{ steps.tfswitch-install.outputs.terraform_version }} installed to $(which terraform) .."
-          ver=${{ steps.tfswitch-install.outputs.terraform_version }}
         else
           echo "Output variable is empty"
           exit 1

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -63,3 +63,18 @@ jobs:
           EOF
           tfswitch
           terraform --version
+
+  export-version-as-output:
+    outputs:
+      terraform_version: ${{ steps.tfswitch-install.outputs.terraform_version }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup tfswitch
+      uses: ./
+      id: tfswitch-install
+    - name: Install
+      run: |
+        tfswitch 1.7.0
+        terraform --version
+        echo "Terraform ${{steps.tfswitch-install.outputs.terraform_version }} installed to $(which terraform) .."

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -5,67 +5,64 @@ on:
     branches:
       - main
 
-env: 
-  ACTIONS_STEP_DEBUG: true
-
 jobs:
-  # test-terraform-version-file:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v4
-  #     - uses: ./
-  #     - name: test with the version in a .terraform-version file
-  #       run: |
-  #         echo "1.6.3" > .terraform-version
-  #         tfswitch
-  #         terraform --version
+  test-terraform-version-file:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+      - name: test with the version in a .terraform-version file
+        run: |
+          echo "1.6.3" > .terraform-version
+          tfswitch
+          terraform --version
 
-  # test-command-line-version:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v4
-  #     - uses: ./
-  #     - name: test with the version on the command line
-  #       run: |
-  #         tfswitch 1.6.2
-  #         terraform --version
+  test-command-line-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+      - name: test with the version on the command line
+        run: |
+          tfswitch 1.6.2
+          terraform --version
 
-  # test-env-var-version:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v4
-  #     - uses: ./
-  #     - name: test with the version in an env var
-  #       env:
-  #         TF_VERSION: 1.6.2
-  #       run: |
-  #         tfswitch 1.6.2
-  #         terraform --version
+  test-env-var-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+      - name: test with the version in an env var
+        env:
+          TF_VERSION: 1.6.2
+        run: |
+          tfswitch 1.6.2
+          terraform --version
 
-  # test-install-latest-version:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v4
-  #     - uses: ./
-  #     - name: test with the version in an env var
-  #       run: |
-  #         tfswitch -u
-  #         terraform --version
+  test-install-latest-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+      - name: test with the version in an env var
+        run: |
+          tfswitch -u
+          terraform --version
 
-  # test-terraform-tf-file:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v4
-  #     - uses: ./
-  #     - name: test with the version in an env var
-  #       run: |
-  #         cat << EOF > versions.tf
-  #         terraform {
-  #           required_version = "~> 1"
-  #         }
-  #         EOF
-  #         tfswitch
-  #         terraform --version
+  test-terraform-tf-file:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+      - name: test with the version in an env var
+        run: |
+          cat << EOF > versions.tf
+          terraform {
+            required_version = "~> 1"
+          }
+          EOF
+          tfswitch
+          terraform --version
 
   test-export-version-as-output:
     runs-on: ubuntu-latest
@@ -80,11 +77,9 @@ jobs:
       run: |
         tfswitch 1.6.6
         terraform --version
-        echo "Output of terraform_version${{ steps.tfswitch-install.outputs.terraform_version }}"
         if [[ -n "${{ steps.tfswitch-install.outputs.terraform_version }}" ]]; then
-          echo "Output variable is set"
           echo "Terraform ${{ steps.tfswitch-install.outputs.terraform_version }} installed to $(which terraform) .."
         else
-          echo "Output variable is empty"
+          echo "Error - expected action output variable is empty"
           exit 1
         fi

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -5,6 +5,9 @@ on:
     branches:
       - main
 
+env: 
+  ACTIONS_STEP_DEBUG: true
+
 jobs:
   # test-terraform-version-file:
   #   runs-on: ubuntu-latest

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -6,68 +6,68 @@ on:
       - main
 
 jobs:
-  test-terraform-version-file:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./
-      - name: test with the version in a .terraform-version file
-        run: |
-          echo "1.6.3" > .terraform-version
-          tfswitch
-          terraform --version
+  # test-terraform-version-file:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: ./
+  #     - name: test with the version in a .terraform-version file
+  #       run: |
+  #         echo "1.6.3" > .terraform-version
+  #         tfswitch
+  #         terraform --version
 
-  test-command-line-version:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./
-      - name: test with the version on the command line
-        run: |
-          tfswitch 1.6.2
-          terraform --version
+  # test-command-line-version:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: ./
+  #     - name: test with the version on the command line
+  #       run: |
+  #         tfswitch 1.6.2
+  #         terraform --version
 
-  test-env-var-version:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./
-      - name: test with the version in an env var
-        env:
-          TF_VERSION: 1.6.2
-        run: |
-          tfswitch 1.6.2
-          terraform --version
+  # test-env-var-version:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: ./
+  #     - name: test with the version in an env var
+  #       env:
+  #         TF_VERSION: 1.6.2
+  #       run: |
+  #         tfswitch 1.6.2
+  #         terraform --version
 
-  test-install-latest-version:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./
-      - name: test with the version in an env var
-        run: |
-          tfswitch -u
-          terraform --version
+  # test-install-latest-version:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: ./
+  #     - name: test with the version in an env var
+  #       run: |
+  #         tfswitch -u
+  #         terraform --version
 
-  test-terraform-tf-file:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./
-      - name: test with the version in an env var
-        run: |
-          cat << EOF > versions.tf
-          terraform {
-            required_version = "~> 1"
-          }
-          EOF
-          tfswitch
-          terraform --version
+  # test-terraform-tf-file:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: ./
+  #     - name: test with the version in an env var
+  #       run: |
+  #         cat << EOF > versions.tf
+  #         terraform {
+  #           required_version = "~> 1"
+  #         }
+  #         EOF
+  #         tfswitch
+  #         terraform --version
 
-  export-version-as-output:
+  test-export-version-as-output:
+    runs-on: ubuntu-latest
     outputs:
       terraform_version: ${{ steps.tfswitch-install.outputs.terraform_version }}
-    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - name: Setup tfswitch
@@ -75,6 +75,13 @@ jobs:
       id: tfswitch-install
     - name: Install
       run: |
-        tfswitch 1.7.0
+        tfswitch 1.6.6
         terraform --version
-        echo "Terraform ${{steps.tfswitch-install.outputs.terraform_version }} installed to $(which terraform) .."
+        if [[ -n "$ver" ]]; then
+          echo "Output variable is set"
+          echo "Terraform ${{ steps.tfswitch-install.outputs.terraform_version }} installed to $(which terraform) .."
+          ver=${{ steps.tfswitch-install.outputs.terraform_version }}
+        else
+          echo "Output variable is empty"
+          exit 1
+        fi

--- a/README.md
+++ b/README.md
@@ -105,6 +105,12 @@ jobs:
         echo "Terraform ${{steps.tfswitch-install.outputs.terraform_version }} installed to $(which terraform) .."
 ```
 
+## Outputs
+
+| **Output**        | **Type** | **Description**                    |
+|-------------------|----------|------------------------------------|
+| terraform_version | string   | The version of terraform installed |
+
 ## Note
 
 This action unlinks the default installed terraform which is available in the [runner](https://github.com/actions/runner-images/tree/main).

--- a/README.md
+++ b/README.md
@@ -82,6 +82,29 @@ jobs:
         ...
 ```
 
+### Use specified version in terraform setup
+
+```yaml
+jobs:
+  export-version-as-output:
+    outputs:
+      terraform_version: ${{ steps.tfswitch-install.outputs.terraform_version }}  
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup tfswitch
+      uses: stv-io/action-tfswitch@v1
+    - name: Install
+      id: tfswitch-install
+      run: |
+        tfswitch 1.6.4
+        terraform --version
+        ...
+    - name: display version and path
+      run: |
+        echo "Terraform ${{steps.tfswitch-install.outputs.terraform_version }} installed to $(which terraform) .."
+```
+
 ## Note
 
 This action unlinks the default installed terraform which is available in the [runner](https://github.com/actions/runner-images/tree/main).

--- a/action.yml
+++ b/action.yml
@@ -4,6 +4,11 @@ author: Stephen Attard
 
 description: Setup tfswitch to install a specific version of terraform
 
+outputs:
+  terraform_version:
+    description: "The installed version of terraform"
+    value: ${{ steps.tfswitch-install.outputs.terraform_version }}
+
 runs:
   using: composite
   steps:
@@ -15,6 +20,7 @@ runs:
         exit 1
     - name: Install tfswitch
       shell: bash
+      id: tfswitch-install
       run: |
         echo "Downloading install script .."
         curl -sLo install.sh https://raw.githubusercontent.com/warrensbox/terraform-switcher/release/install.sh
@@ -33,11 +39,10 @@ runs:
         echo $HOME/.bin >> $GITHUB_PATH
         echo "Exporting terraform version to GITHUB_OUTPUT as terraform_version.."
         echo "terraform_version=$(terraform --version -json | jq -r .terraform_version)" >> "$GITHUB_OUTPUT"
-        echo "Debug - dump github ouput"
-        echo $GITHUB_OUTPUT
-        cat $GITHUB_OUTPUT
-        echo "Unlinking default version of terraform .."
-        unlink /usr/local/bin/terraform
+        if [ -e "/usr/local/bin/terraform" ]; then
+            echo "Unlinking default version of terraform .."
+            unlink /usr/local/bin/terraform
+        fi
         echo "Done!"
 
 branding:

--- a/action.yml
+++ b/action.yml
@@ -35,6 +35,7 @@ runs:
         echo "terraform_version=$(terraform --version -json | jq -r .terraform_version)" >> "$GITHUB_OUTPUT"
         echo "Debug - dump github ouput"
         echo $GITHUB_OUTPUT
+        cat $GITHUB_OUTPUT
         echo "Unlinking default version of terraform .."
         unlink /usr/local/bin/terraform
         echo "Done!"

--- a/action.yml
+++ b/action.yml
@@ -32,9 +32,9 @@ runs:
         echo "Exporting PATH to GITHUB_PATH .."
         echo $HOME/.bin >> $GITHUB_PATH
         echo "Exporting terraform version to GITHUB_OUTPUT as terraform_version.."
-        echo "DEBUG VERSION"
-        terraform --version -json | jq -r .terraform_version
         echo "terraform_version=$(terraform --version -json | jq -r .terraform_version)" >> "$GITHUB_OUTPUT"
+        echo "Debug - dump github ouput"
+        echo $GITHUB_OUTPUT
         echo "Unlinking default version of terraform .."
         unlink /usr/local/bin/terraform
         echo "Done!"

--- a/action.yml
+++ b/action.yml
@@ -32,6 +32,8 @@ runs:
         echo "Exporting PATH to GITHUB_PATH .."
         echo $HOME/.bin >> $GITHUB_PATH
         echo "Exporting terraform version to GITHUB_OUTPUT as terraform_version.."
+        echo "DEBUG VERSION"
+        terraform --version -json | jq -r .terraform_version
         echo "terraform_version=$(terraform --version -json | jq -r .terraform_version)" >> "$GITHUB_OUTPUT"
         echo "Unlinking default version of terraform .."
         unlink /usr/local/bin/terraform

--- a/action.yml
+++ b/action.yml
@@ -17,7 +17,7 @@ runs:
       shell: bash
       run: |
         echo "Downloading install script .."
-        curl -sLo install.sh https://raw.githubusercontent.com/warrensbox/terraform-switcher/release/install.sh 
+        curl -sLo install.sh https://raw.githubusercontent.com/warrensbox/terraform-switcher/release/install.sh
         echo "Configuring install script to be executable .."
         chmod 755 install.sh
         echo "Creating $HOME/.bin for tfswitch binary .."
@@ -31,6 +31,8 @@ runs:
         tfswitch --version
         echo "Exporting PATH to GITHUB_PATH .."
         echo $HOME/.bin >> $GITHUB_PATH
+        echo "Exporting terraform version to GITHUB_OUTPUT as terraform_version.."
+        echo "terraform_version=$(terraform --version -json | jq -r .terraform_version)" >> "$GITHUB_OUTPUT"
         echo "Unlinking default version of terraform .."
         unlink /usr/local/bin/terraform
         echo "Done!"

